### PR TITLE
Rework secret-handshake exercise docs

### PR DESCRIPTION
**Pull request by kytrinyx**
From ChatGPT:

Story: Have you ever heard the joke about "there are 10 types of people in the world"? Well, there are those who understand binary, and those who don't. You belong to the former group and you and your binary-savvy friends want to create a secret handshake. You don't want anyone who isn't in the know to be able to crack the code. So you come up with a sequence of actions, where each action corresponds to a different binary digit.
Instructions: Your task is to write a program that converts a decimal number to a sequence of actions in the secret handshake.
The secret handshake consists of 5 different actions. The first action corresponds to the least significant bit (2^0) and the fifth action corresponds to the most significant bit (2^4).
If the decimal number has a binary digit of 1 in the least significant bit (2^0), the corresponding action is "wink".
If it has a binary digit of 1 in the second least significant bit (2^1), the corresponding action is "double blink".
If it has a binary digit of 1 in the third least significant bit (2^2), the corresponding action is "close your eyes".
If it has a binary digit of 1 in the fourth least significant bit (2^3), the corresponding action is "jump".
If it has a binary digit of 1 in the most significant bit (2^4), the actions should be reversed.
For example, if the decimal number is 19, which is 16+2+1 in powers of two and thus 10011 in binary, then the corresponding sequence of actions is ["double blink", "wink"].

### CODE REVIEW

File: exercises/secret-handshake/introduction.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+There's a joke that goes:
+
+_There are 10 types of people in the world._
+_There are those who understand binary, and those who don't._
+
+You belong to the group that understands binary, and you and your binary-savvy friends want to create a secret handshake.
+You don't want anyone who isn't in the know to be able to crack the code.
+
+So you come up with a sequence of actions, where each action corresponds to a different binary digit.
```
Comment (iHiD) [inline exercises/secret-handshake/introduction.md:11]: Can we add a exercism/note that says "No idea what binary is or how to use it? Check this tutorial..."

Comment (iHiD) [inline exercises/secret-handshake/introduction.md:11]: I feel like binary is an implementation detail, not the story. And so I feel like the joke is maybe unnecessary and adds noise. So I suggest getting rid of that and focussing a little more on the fun story angle.

Suggestion:

```suggestion
You are making a secret coding club with some friends and friends-of-friends. 
You don't know everyone in the club, so you and your friends have decided to create a secret handshake that you can use to recognise that someone is a member.
You don't want anyone who isn't in the know to be able to crack the code.

You and your friends have decided to come up with a code that use binary (because you're a geek and think binary is cool) and turns it into a series of actions that you do during the handshake.
```

File: exercises/secret-handshake/instructions.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# Instructions
+
+Your task is to converts a number between 1 and 31 to a sequence of actions in the secret handshake.
+
+The sequence of actions are chosen by looking at each of the number places in the binary number.
+Starting at the back, looking at the digit that is farthest to the right.
+Then, move left.
```
Comment (iHiD) [inline exercises/secret-handshake/instructions.md:7]: Reference to "number place" and "back" confused me.

```suggestion
The sequence of actions are chosen by looking at each of the digits in the binary number.
Start at the right-most digit and move left.
```

Comment (kytrinyx) [reply exercises/secret-handshake/instructions.md:7]: Cool. I was trying to figure out how to avoid "least significant digit" 😅

File: exercises/secret-handshake/introduction.md (hunk 1 above)
Comment (kytrinyx) [reply exercises/secret-handshake/introduction.md:11]: Nice, yeah. It's so funny how I even failed to consider that we could dump the binary thing.

Comment (kytrinyx) [reply exercises/secret-handshake/introduction.md]: I searched for something by Vaidehi, and found this: https://medium.com/basecs/bits-bytes-building-with-binary-13cb4289aafa

Look good?

File: exercises/secret-handshake/instructions.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+Going left, the next digit is 0, so there is no double-blink.
+Going left again, the next digit is 0, so you leave your eyes open.
+Going left again, the next digit is 1, so you jump.
+That was the last digit, so the final code is:
+
+```
+wink, jump
+```
+
+Given the number 26, which is 11010 in binary, we get the following actions:
```
Comment (ErikSchierboom) [inline exercises/secret-handshake/instructions.md:31]: In the above example, `1001` is put in backticks. Do we want that here too?

File: exercises/secret-handshake/introduction.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
+# Introduction
+
+You are making a secret coding club with some friends and friends-of-friends.
+You don't know everyone in the club, so you and your friends have decided to create a secret handshake that you can use to recognise that someone is a member.
+You don't want anyone who isn't in the know to be able to crack the code.
+
+The secret handshake is created by taking a binary number and turning it into a series of actions.
```
Comment (ErikSchierboom) [inline exercises/secret-handshake/introduction.md:7]: "is created by taking a binary number"

Maybe rephrase slightly to make it seem more, eh, designed? As in: you and friends have designed a binary coding scheme or something like that.

File: exercises/secret-handshake/instructions.md (hunk 3 above)
Comment (kytrinyx) [reply exercises/secret-handshake/instructions.md]: Yes, good call.

File: exercises/secret-handshake/introduction.md [hunk 5]
```diff
@@ -0,0 +1,12 @@
+# Introduction
+
+You are making a secret coding club with some friends and friends-of-friends.
```
Comment (iHiD) [inline exercises/secret-handshake/introduction.md:3]:
```suggestion
You are starting a secret coding club with some friends and friends-of-friends.
```

File: exercises/secret-handshake/introduction.md [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+You don't know everyone in the club, so you and your friends have decided to create a secret handshake that you can use to recognise that someone is a member.
+You don't want anyone who isn't in the know to be able to crack the code.
+
+You designed the code so that it takes a binary number and turns it into a series of actions.
+
+```exercism/note
+If you aren't sure how binary works, check out [this tutorial][intro-to-binary].
```
Comment (iHiD) [inline exercises/secret-handshake/introduction.md:10]:
```suggestion
If you aren't sure what binary is or how it works, check out [this tutorial][intro-to-binary].
```

File: exercises/secret-handshake/instructions.md [hunk 7]
```diff
@@ -0,0 +1,40 @@
+# Instructions
+
+Your task is to converts a number between 1 and 31 to a sequence of actions in the secret handshake.
+
+The sequence of actions are chosen by looking at each of the digits in the binary number.
```
Comment (iHiD) [inline exercises/secret-handshake/instructions.md:5]:
```suggestion
The sequence of actions are chosen by looking at each digit of the number once it's been converted to binary.
```

File: exercises/secret-handshake/instructions.md [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+```
+
+Let's use the number 9 as an example.
+9 in binary is `1001`.
+
+The digit that is farthest to the right is 1, so the first action is `wink`.
+Going left, the next digit is 0, so there is no double-blink.
+Going left again, the next digit is 0, so you leave your eyes open.
+Going left again, the next digit is 1, so you jump.
+That was the last digit, so the final code is:
```
Comment (iHiD) [inline exercises/secret-handshake/instructions.md:25]:
```suggestion
Let's use the number `9` as an example:

- 9 in binary is `1001`.
- The digit that is farthest to the right is 1, so the first action is `wink`.
- Going left, the next digit is 0, so there is no double-blink.
- Going left again, the next digit is 0, so you leave your eyes open.
- Going left again, the next digit is 1, so you jump.

That was the last digit, so the final code is:
```

File: exercises/secret-handshake/introduction.md [hunk 9]
```diff
@@ -0,0 +1,12 @@
+# Introduction
+
+You are making a secret coding club with some friends and friends-of-friends.
+You don't know everyone in the club, so you and your friends have decided to create a secret handshake that you can use to recognise that someone is a member.
+You don't want anyone who isn't in the know to be able to crack the code.
+
+You designed the code so that it takes a binary number and turns it into a series of actions.
```
Comment (iHiD) [inline exercises/secret-handshake/introduction.md:7]:
```suggestion
You designed the code so that one person says a number between 1 and 30, and the other other person turns it into a series of actions.
```

File: exercises/secret-handshake/introduction.md (hunk 6 above)
Comment (iHiD) [reply exercises/secret-handshake/introduction.md:10]: Suggest moving this to instructions, and removing all reference to binary out of introduction.

File: exercises/secret-handshake/introduction.md (hunk 9 above)
Comment (kytrinyx) [reply exercises/secret-handshake/introduction.md:7]:
```
> 31.to_s(2)
=> "11111"
> 30.to_s(2)
=> "11110"
```

Comment (kytrinyx) [reply exercises/secret-handshake/introduction.md]: Fixed this.

File: exercises/secret-handshake/introduction.md [hunk 10]
```diff
@@ -0,0 +1,7 @@
+# Introduction
+
+You are starting a secret coding club with some friends and friends-of-friends.
+You don't know everyone in the club, so you and your friends have decided to create a secret handshake that you can use to recognise that someone is a member.
+You don't want anyone who isn't in the know to be able to crack the code.
+
+You designed the code so that one person says a number between 1 and 31, and the other other person turns it into a series of actions.
```
Comment (iHiD) [inline exercises/secret-handshake/introduction.md:7]:
```suggestion
You've designed the code so that one person says a number between 1 and 31, and the other other person turns it into a series of actions.
```

File: exercises/secret-handshake/introduction.md [hunk 11]
```diff
@@ -0,0 +1,7 @@
+# Introduction
+
+You are starting a secret coding club with some friends and friends-of-friends.
+You don't know everyone in the club, so you and your friends have decided to create a secret handshake that you can use to recognise that someone is a member.
```
Comment (iHiD) [inline exercises/secret-handshake/introduction.md:4]: Just one nit to stop every sentence starting with "you"

```suggestion
Not everyone knows each other, so you and your friends have decided to create a secret handshake that you can use to recognise that someone is a member.
```

**Review by iHiD — APPROVED** [review]

File: exercises/secret-handshake/introduction.md [hunk 12]
```diff
@@ -0,0 +1,7 @@
+# Introduction
+
+You are starting a secret coding club with some friends and friends-of-friends.
+Not everyone knows each other, so you and your friends have decided to create a secret handshake that you can use to recognise that someone is a member.
+You don't want anyone who isn't in the know to be able to crack the code.
+
+You've designed the code so that one person says a number between 1 and 31, and the other other person turns it into a series of actions.
```
Comment (ErikSchierboom) [inline exercises/secret-handshake/introduction.md:7]: I like this new version! We no longer mention anything about using a bit-wise encoding scheme though. That was intentional?

File: exercises/secret-handshake/instructions.md [hunk 13]
```diff
@@ -0,0 +1,46 @@
+# Instructions
+
+Your task is to converts a number between 1 and 31 to a sequence of actions in the secret handshake.
```
Comment (ErikSchierboom) [inline exercises/secret-handshake/instructions.md:3]:
```suggestion
Your task is to convert a number between 1 and 31 to a sequence of actions in the secret handshake.
```

File: exercises/secret-handshake/instructions.md [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+
+The sequence of actions are chosen by looking at each digit of the number once it's been converted to binary.
```
Comment (ErikSchierboom) [inline exercises/secret-handshake/instructions.md:5]: Maybe be explicit about which digits to check?

```suggestion
The sequence of actions are chosen by looking at the rightmost five digits of the number once it's been converted to binary.
```

**Review by ErikSchierboom — APPROVED** [review]

File: exercises/secret-handshake/introduction.md (hunk 12 above)
Comment (kytrinyx) [reply exercises/secret-handshake/introduction.md:7]: Yeah, we moved all the bits (no pun intended) about binary into the instructions, leaving the introduction only as the scenario that puts the exercise into context.

File: exercises/secret-handshake/instructions.md (hunk 14 above)
Comment (kytrinyx) [reply exercises/secret-handshake/instructions.md:5]: The largest number in the range is 5 digits long in binary, so we never have more digits than that.

Comment (ErikSchierboom) [reply exercises/secret-handshake/instructions.md:5]: True, but I could imagine programmers thinking about numbers in terms of bytes and multiples of 8 bits.

Comment (kytrinyx) [reply exercises/secret-handshake/instructions.md:5]: Yes, that's a good point.

Comment (kytrinyx) [reply exercises/secret-handshake/instructions.md:5]: I'm going to add this, and then maybe someone in the problem-specs repo will have some other good idea.

**Review by ErikSchierboom — APPROVED** [review]
